### ISSUE TITLE
Handle deprecated keystone module

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt#49419]
- [x] salt/modules/keystone.py

The keystone module has been replaced with keystoneng.py.

Don't forget to remove the autodoc files + references, as well as the stale doc files on the doc build server.

**Comment by Ch3LL** [comment]
blocked by https://github.com/saltstack/salt/issues/55611

**Comment by sagetherage** [comment]
closing this in favor of https://github.com/saltstack/salt/issues/58032
